iree/vm/shims.h

```diff
@@ -0,0 +1,83 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IREE_VM_TEST_EMITC_SHIMS_H_
+#define IREE_VM_TEST_EMITC_SHIMS_H_
+
+#include "iree/vm/module.h"
+#include "iree/vm/stack.h"
+
+// see Calling convetion in module.h
+// We use the same format but replace '.' by '_'
+// Variadic arguments are not supported
+
+// 0.
```

Comment:
As a bit of follow-on work these shims can be generated and kept in the output module file - that way each module generates what it needs and there's no need for manual editing of the code as new signatures are required. The linker will take care of deduping the actual code across all modules that may be linked in just like how it will have to here because these are all in .h's and instantiated in every compilation unit that uses them.

My plan is to also add an iree-tblgen generator (or something) that can take vm.import ops and produce the shim functions for user-authored custom modules. Then hopefully users writing their own modules wouldn't have to hand-write/select these shims and combined with the above neither would us maintainers as the emitc target would need to add more!

Reply:
Sure, I didn't want to pack everything into the printer. I have to think about how to keep this / make this more readable.